### ISSUE TITLE
adding port forwarding directive breaks the SSH port

### ISSUE THREAD

**Issue by hzj112** [hashicorp/vagrant#12047]
Sorry for raising the same issue, if already reported. But I did do a search, I couldn't find any similar issues being reported. 

When I include a port forwarding directive in the Vagrantfile (`config.vm.network :forwarded_port, guest: 80, host:8080`), the ssh address port portion is always set to 22, however, it is actually ported to 2222 on the host. If I removed that line, then `vagrant up` will run through fine. 

I have tried to add directive to explicitly forwarding ssh port 22 to a value such as 2222, but getting the same error/warning `Warning: Authentication failure. Retrying...`. 

```
==> node1: Preparing network interfaces based on configuration...
    node1: Adapter 1: nat
    node1: Adapter 2: hostonly
==> node1: Forwarding ports...
    node1: 80 (guest) => 8080 (host) (adapter 1)
    node1: 22 (guest) => 2222 (host) (adapter 1)
==> node1: Running 'pre-boot' VM customizations...
==> node1: Booting VM...
==> node1: Waiting for machine to boot. This may take a few minutes...
    node1: SSH address: 127.0.0.1:22
    node1: SSH username: vagrant
    node1: SSH auth method: private key
    node1: Warning: Authentication failure. Retrying...
    node1: Warning: Authentication failure. Retrying...
```

`vagrant ssh-config` shows the following

```
vagrant ssh-config
Host node1
  HostName 127.0.0.1
  User vagrant
  Port 22
  UserKnownHostsFile /dev/null
  StrictHostKeyChecking no
  PasswordAuthentication no
  IdentityFile /Users/myuser/.vagrant.d/insecure_private_key
  IdentitiesOnly yes
  LogLevel FATAL
```

if I ran `ssh vagrant@localhost -p 2222`, it can get me into the box, so the port is forwarded to 2222. 

This also happens to different boxes, such as centos/7, so it probably not an issue with box. 

Thanks in advance!

### Vagrant version

RVagrant 2.2.13

### Host operating system

Mac OS 11.0.1 (20B29) Big Sur

### Guest operating system

BOX = "bento/ubuntu-18.04"
BOX_VERSION = "202010.24.0"

### Vagrantfile

```
# -*- mode: ruby -*-
# # vi: set ft=ruby :

require "./subnet.rb"

# Every Vagrant development environment requires a box. You can search for
BOX = "bento/ubuntu-18.04"
BOX_VERSION = "202010.24.0"

nodes = [
#    {:name => :<node-name>, :cpu => 2, :mem => 4096, :ip => "%s.<forth-quarter-ip>" % [load_subnet]},
     {:name => :node1, :cpu => 2, :mem => 4096, :ip => "%s.201" % [load_subnet], :ssh_port => 2201}
#     {:name => :node2, :cpu => 2, :mem => 4096, :ip => "%s.202" % [load_subnet], :ssh_port => 2202},
]

VAGRANTFILE_API_VERSION = "2"

HOST_TLD = "myhost"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|

  config.vm.box = BOX
  config.vm.box_version = BOX_VERSION

  nodes.each do |opts|
    config.vm.define opts[:name] do |config|
        config.vm.hostname = "%s.%s" % [opts[:name].to_s, HOST_TLD]
        config.vm.network :private_network, ip: opts[:ip]
        config.vm.network :forwarded_port, guest: 80, host:8080 # if comment this line, ssh address port is set correct
        config.vm.provider :virtualbox do |vb|
            vb.customize ["modifyvm", :id, "--cpus", opts[:cpu] || 4 ]
            vb.customize ["modifyvm", :id, "--memory", opts[:mem] || 8192 ]
        end
        config.vm.synced_folder ".", "/vagrant", type: "virtualbox", disabled: false
    end
  end
end
```

### Debug output

Provide a link to a GitHub Gist containing the complete debug output:
https://www.vagrantup.com/docs/other/debugging.html. The debug output should
be very long. Do NOT paste the debug output in the issue, just paste the
link to the Gist.

### Expected behavior

SSH Address, port should be the same value as ported value

### Actual behavior

SSH Address port always set to 22

### Steps to reproduce

`vagrant up`

```
Bringing machine 'node1' up with 'virtualbox' provider...
==> node1: Importing base box 'bento/ubuntu-18.04'...
==> node1: Matching MAC address for NAT networking...
==> node1: Checking if box 'bento/ubuntu-18.04' version '202010.24.0' is up to date...
==> node1: Setting the name of the VM: ubuntu-station_node1_1605373395363_3208
==> node1: Clearing any previously set network interfaces...
==> node1: Preparing network interfaces based on configuration...
    node1: Adapter 1: nat
    node1: Adapter 2: hostonly
==> node1: Forwarding ports...
    node1: 80 (guest) => 8080 (host) (adapter 1)
    node1: 22 (guest) => 2222 (host) (adapter 1)
==> node1: Running 'pre-boot' VM customizations...
==> node1: Booting VM...
==> node1: Waiting for machine to boot. This may take a few minutes...
    node1: SSH address: 127.0.0.1:22
    node1: SSH username: vagrant
    node1: SSH auth method: private key
    node1: Warning: Authentication failure. Retrying...
    node1: Warning: Authentication failure. Retrying...
```

**Comment by sengfai** [comment]
1. Update your virtualbox
2. Update your vagrant
3. Go to System Preference then privacy& security and Allow the permission in general tap.

**Comment by soapy1** [comment]
Hey there, this seems to be the same issue as https://github.com/hashicorp/vagrant/issues/12045. The issue has a [more detailed explanation](https://github.com/hashicorp/vagrant/issues/12045#issuecomment-726902176) of the problem. 

A workaround for this is to enable connecting to the guest via the assigned IP address directly using ssh_info_public:

```
Vagrant.configure("2") do |config|
  config.vm.provider :vmware_desktop do |v|
    v.ssh_info_public = true
  end
end
```

**Comment by lukepolo** [comment]
Its slightly different in that we are referring to **virtualbox** . Just want to make sure that's highlighted.

**Comment by soapy1** [comment]
aha! thanks for that @lukepolo, opening this back up

**Comment by lukepolo** [comment]
Quick hack to open SSL tunnels : note it does not keep track of already opened tunnels 

```
#!/usr/bin/env bash
ports=(
  "80, 8080"
  9000
)

numberRx='^[0-9]+$'

for port in "${ports[@]}"
do
   :
  portForward=$port
  if ! [[ $port =~ $numberRx ]] ; then
    IFS=', ' read -r -a temp <<< "$port"

    portForward=${temp[1]}
    ssh -f -N -L $portForward:192.168.10.10:$port vagrant@192.168.10.10
  else
    ssh -f -N -L $port:192.168.10.10:$port vagrant@192.168.10.10
  fi
  echo "Forwarding $portForward to $port";
done
```

**Comment by chrisroberts** [comment]
Hi,

I have so far been unable to reproduce the behavior which has been described. Would you please provide a debug gist for the the `ssh-config` command (`vagrant ssh-config --debug`). Other information that would be helpful:

* Version of VirtualBox installed
* Did guest VMs exist prior to upgrade to Big Sur?
* Do you see the same behavior with a minimal Vagrantfile?

For the last one, an example would be:

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "bento/ubuntu-18.04"
  config.vm.network :forwarded_port, guest: 80, host: 8080
end
```

**Comment by hzj112** [comment]
> vagrant ssh-config --debug

hi @chrisroberts  please see https://gist.github.com/hzj112/9b78e8edc4bd51de85f826173cba7098 for the information you asked about. 



Version of VirtualBox installed

2.2.13

Did guest VMs exist prior to upgrade to Big Sur?
no, they are new VMs. 

Do you see the same behavior with a minimal Vagrantfile? 
Yes, same behavior. FYI, in the minimal Vagrantfile I changed host port to 8888, since I have a service running on 8080 now on my host.  I uploaded relevant information in the gist above.

**Comment by chrisroberts** [comment]
@hzj112 Fantastic, thank you!

**Comment by chrisroberts** [comment]
@hzj112 Based on the version of VirtualBox you have installed (6.1.97) you are currently using a development build of VirtualBox. This version does not appear to include forwarding information when inspecting a guest, which results in invalid results. Since the root cause of this error is originating from a development build and not a proper release, the best course of action to resolve the issue is to uninstall your current version of VirtualBox and install the latest stable release.

Cheers!

**Comment by lukepolo** [comment]
I have this same problem with latest virutalbox

![image](https://user-images.githubusercontent.com/2066668/99568771-c9f7da00-299d-11eb-8461-e8cc5728358d.png)

**Comment by chrisroberts** [comment]
@lukepolo If you are encountering the same behavior, would you please provide a gist of the debug output from a `vagrant ssh-config --debug` command so we can determine the underlying cause and if it is the same as the initially reported issue.

Thanks

**Comment by lavjamanxd** [comment]
I am facing the same issue, i have defined multiple machines in a Vagrant file.
Somehow on the first machine it can detect the correct port for ssh automatically,
but for the other ones, it tries to connect to a wrong port on localhost.

As you can see in your output: "node1: SSH address: 127.0.0.1:22" thats your local machines ssh and not the virtualboxes fowarded ssh port.

So for a workaround you can manually specify the port forward for 22 => <someport> then set the ssh-s port explicitly to that <someport>

in your case:

```
node1.vm.network "forwarded_port", guest: 22, host: 3200, id: 'ssh'
node1.ssh.port = 3202
```

**Comment by 4levels** [comment]
Comment from related issue #12045 - the title didn't mention VMWare at the time of posting and this issue was already closed.

----------------------------
We just experienced a very similar issue with the Virtualbox provider.  Vagrant never manages to connect but connecting via ssh just works.
This issue prevents us from running our provision steps.
Given that #12047 was already closed I added my comment here.

FYI we did update everything to the latest version and made sure the OSX permissions were all set. We rebooted as well to make sure the kernel modules were loaded correctly.

We have now resorted to downgrading back to Catalina so I cannot currently reproduce or run a debug session.

I sincerely hope by the time we all upgrade to Big Sur the transition will be smoother.

----------------------------

IMHO this is still broken and despite a manual workaround exists, this is still an ongoing issue and therefor should remain open.  As explained before, we neither have time or the knowledge to debug this and resorted to reverting to the previous OSX version so we cannot even reproduce anymore.
